src/visualization/src/IrrlichtUtils.h

```diff
@@ -160,6 +162,8 @@ inline irr::scene::ISceneNode * addGeometryToSceneManager(const iDynTree::SolidS
 
         geomNode = smgr->addMeshSceneNode(boxMesh,linkNode);
 
+        boxMesh->drop();
```

Comment:
How did you realized that this was necessary?

Reply:
Using valgrind. To be honest the way irrlicht deals with this sort of smart pointers is quite blurry to me. Looking at some examples online I noticed they where dropping the pointer to the mesh after adding it to the scene. As you noticed, I did a lot of experiments before avoiding any problem with valgrind 😁